VERSIONING.md

```diff
@@ -0,0 +1,54 @@
+
+# Versioning sofware in the TDP Project
+ 
+This document is to provide guidelines for versioning and releasing software in the TDP Project.
+
+As of April 2023, TDP project includes:
+
+- Stack containing Apache Hadoop Softwares along with building instructions and patches backported by the TOSIT. ([TOSIT-IO/hadoop](https://github.com/TOSIT-IO/hadoop), [TOSIT-IO/hive](https://github.com/TOSIT-IO/hive) ...)
+- Ansible collection which installs said software. ([TOSIT-IO/tdp-collection](https://github.com/TOSIT-IO/tdp-collection))
+- Library which installs Ansible collections exposing a DAG. ([TOSIT-IO/tdp-lib](https://github.com/TOSIT-IO/tdp-lib))
+  - A HTTP Server and web UI are built on top of this library. ([TOSIT-IO/tdp-server](https://github.com/TOSIT-IO/tdp-server), [TOSIT-IO/tdp-ui](https://github.com/TOSIT-IO/tdp-ui))
+
+## TDP Stack
+
+TDP Stack expose a version number that creates a coherent state of its software.
+
+The Stack is composed of 2 digits
+- The first one is the major version and should be bumped in the event of a major component change with breaking changes (i.e: Hadoop major version upgrade)
+- The second one is the minor version and should be bumped when a component version updates for security and/or new features with no or minor breaking changes
+
+Each software in the stack contains the Apache version it was based plus an appendix reserved for TOSIT users.
+
+For example:
+
+- TDP Stack in version `1.1` includes HBase forked from `2.1.10`
+- HBase version is modified by TOSIT to `2.1.10-1.0`. In the appendix `-1.0`:
+  - The first digit is bumped by TOSIT and contains bugfix and/or backport.
+  - The second digit is NEVER modified by TOSIT. It is there for any user and vendor to create non cumulative patch to the software. It allows anyone to create its own release of a product without requiring approval from TOSIT association. This patch number is not necessarily cumulative. If appropriate, the content of said patch should be included in the TOSIT repository in the form of a patch that would bump the first digit of the appendix.
+
+Unreleased changes should be tagged `-SNAPSHOT` as per Java Standard.
+
+Upgrading the stack for security bumps the last digit of the stack. This upgrade can include an Apache upstream security path and/or a TOSIT patch. 
+
+## TDP Collection
+
+TDP-Collection is tightly coupled to the TDP Stack it installs.
+
+As an example TDP-Collection 1.0 installs TDP Stack `1.1`. When the stack upgrades to `1.2`, it requires collection to upgrade to `2.0`.
```

Comment:
If we update the TDP stack to `2.1` from `1.5`, the collection is in version `5.0` and it will change to `1.0`? I don't think since that would make a version downgrade.

Reply:
This is true indeed and is one caveat of this policy. I think this needs to be put into perspective:

- I don't think TDP Stack 2.X is going to happen anytime soon.
- If it were to happen I see multiple ways to solve the issue:
  - Assume that tdp-collection is only meant to deploy TDP-Stack 1.X and create a new project which deploy TDP Stack 2.X
  - Assume that major stack upgrade of the stack can be done as a collection major upgrade
    - If we update the stack to `2.1` from `1.5`, this upgrade is reflected in the upgrade of collection from `5.0` to `6.0`.
  - Add a digit to tdp-collection reflecting what major stack version is meant to deploy (i.e: tdp-collection 1.1.X deploys tdp-stack 1.1, collection 2.3.X deploys stack 2.3).

My opinion is that the simplicity of the versioning is valuable and that I do not know what TDP-Stack 2.X will look like. I also think that the versioning policy can evolve and that is not a big deal (if it does not happen every other week ofc).

Arguments can be made that for the sake of completeness we should cover all future use cases. This can be discussed.

Reply:
As discuss, It is unclear what TDP 2.0 will be.